HT-242 Save and Merge Block Skip data in HearThisPacks

JohnThomson opened the pull request.

tombogle commented:
*[src/HearThisTests/ClipRepositoryCharacterFilterTests.cs, line 173 at r1](https://reviewable.io/reviews/sillsdev/hearthis/99#-LC5Tk9tClQFixlO3rcs:-LC5at531vNceJpUZ6MY:b-7kx5dp) ([raw file](https://github.com/sillsdev/hearthis/blob/c93654e82c275e1272b1075c55768969b1526654/src/HearThisTests/ClipRepositoryCharacterFilterTests.cs#L173)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Not needed, right?
</blockquote></details>

If it is, there will be a failing test :-)

---


*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/hearthis/99)*
<!-- Sent from Reviewable.io -->

tombogle commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---







*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/hearthis/99#-:-LC5bBLaAffKro8u70j4:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tombogle commented:
*[src/HearThisTests/ClipRepositoryCharacterFilterTests.cs, line 173 at r1](https://reviewable.io/reviews/sillsdev/hearthis/99#-LC5Tk9tClQFixlO3rcs:-LC5bzxD9AwWOjt6W6gt:b-rg4do3) ([raw file](https://github.com/sillsdev/hearthis/blob/c93654e82c275e1272b1075c55768969b1526654/src/HearThisTests/ClipRepositoryCharacterFilterTests.cs#L173)):*
> If it is, there will be a failing test :-)
Sorry, I was looking at the method above this one.

---


*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/hearthis/99)*
<!-- Sent from Reviewable.io -->

JohnThomson commented:
You caught some minor things, but worth fixing.

---

Review status: 8 of 13 files reviewed at latest revision, 6 unresolved discussions.

---

*[src/HearThis/Script/HearThisPackMaker.cs, line 35 at r1](https://reviewable.io/reviews/sillsdev/hearthis/99#-LC5UFeG6XJx_QLU0u7k:-LC5khFt2KKqFE1N7YP2:bft2byy) ([raw file](https://github.com/sillsdev/hearthis/blob/c93654e82c275e1272b1075c55768969b1526654/src/HearThis/Script/HearThisPackMaker.cs#L35)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Why doesn't this zip up the skip file if it exists?
</blockquote></details>

ZipUp is coded to only include .wav files and a single info.xml file in each directory. I've changed its name to indicate this. Not sure why I coded it that way, since typically there is nothing else there. But some of the logic is to limit things to a single voice actor, and we only know how to do that for wav files.

---

*[src/HearThis/Script/RepoMerger.cs, line 60 at r1](https://reviewable.io/reviews/sillsdev/hearthis/99#-LC5YKNN0sqdbbdtZ933:-LC5lNPS7q5dr0TCbGlX:byly6b6) ([raw file](https://github.com/sillsdev/hearthis/blob/c93654e82c275e1272b1075c55768969b1526654/src/HearThis/Script/RepoMerger.cs#L60)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I thought merging was done now for HearThisPacks, too (not just Android).
</blockquote></details>

Yes, it is, and that's exactly what the current changes are designed to do. But merging from a HearThisPack only modifies what's on the current computer ('mine' or 'ours'). We don't modify the HearThisPack, much less the computer it came from. So propagating changes to 'theirs' is not useful. When 'theirs' is an Android device, it becomes useful.

---

*[src/HearThis/Script/SkippedScriptLines.cs, line 76 at r1](https://reviewable.io/reviews/sillsdev/hearthis/99#-LC5XD86-j4u9dQWZleE:-LC5nCN3Cq_2WpcD59W7:b3yppo3) ([raw file](https://github.com/sillsdev/hearthis/blob/c93654e82c275e1272b1075c55768969b1526654/src/HearThis/Script/SkippedScriptLines.cs#L76)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Should there be a version of IsSameLine which takes bookNumber, etc. and which is called by the one you just created?
</blockquote></details>

YAGNI (unless you see some place I should use the extra method?)

---

*[src/HearThis/Script/SkippedScriptLines.cs, line 108 at r1](https://reviewable.io/reviews/sillsdev/hearthis/99#-LC5WsDp2ZLgl1dY_D4X:-LC5nQvh9z4y0wDrEuvJ:b-896fix) ([raw file](https://github.com/sillsdev/hearthis/blob/c93654e82c275e1272b1075c55768969b1526654/src/HearThis/Script/SkippedScriptLines.cs#L108)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

IsSameLine?
</blockquote></details>

Done.

---

*[src/HearThisTests/ClipRepositoryCharacterFilterTests.cs, line 173 at r1](https://reviewable.io/reviews/sillsdev/hearthis/99#-LC5Tk9tClQFixlO3rcs:-LC5mvgF9kcuW5qpBBmF:b-6cyaqr) ([raw file](https://github.com/sillsdev/hearthis/blob/c93654e82c275e1272b1075c55768969b1526654/src/HearThisTests/ClipRepositoryCharacterFilterTests.cs#L173)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

> If it is, there will be a failing test :-)
Sorry, I was looking at the method above this one.
</blockquote></details>

Removed. It was from an earlier version of a change to the interface.

---


*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/hearthis/99#-:-LC5oLP51Ev0poh62Zl4:bp053a6)*
<!-- Sent from Reviewable.io -->
